coderd/wsbuilder/wsbuilder.go

```diff
@@ -610,67 +663,91 @@ func (b *Builder) getParameters() (names, values []string, err error) {
 		return *b.parameterNames, *b.parameterValues, nil
 	}
 
-	templateVersionParameters, err := b.getTemplateVersionParameters()
+	// Always reject legacy parameters.
+	err = b.verifyNoLegacyParameters()
 	if err != nil {
-		return nil, nil, BuildError{http.StatusInternalServerError, "failed to fetch template version parameters", err}
+		return nil, nil, BuildError{http.StatusBadRequest, "Unable to build workspace with unsupported parameters", err}
+	}
+
+	if b.usingDynamicParameters() {
+		names, values, err = b.getDynamicParameters()
+	} else {
+		names, values, err = b.getClassicParameters()
+	}
```

Comment:
`b.getClassicParameters` is the original logic. The template has to be opted into dynamic params for it to use the new code path. DynamicParameters is currently in Beta